src/main/java/jp/skypencil/jenkins/regression/RegressionReportHelper.java

```diff
@@ -0,0 +1,84 @@
+package jp.skypencil.jenkins.regression;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+import java.util.HashMap;
+
+import hudson.model.AbstractBuild;
+import hudson.tasks.junit.CaseResult;
+import hudson.tasks.junit.ClassResult;
+import hudson.tasks.junit.PackageResult;
+import hudson.tasks.junit.TestResult;
+
+import hudson.tasks.test.AbstractTestResultAction;
+import hudson.tasks.test.AggregatedTestResultAction;
+import hudson.tasks.junit.TestResultAction;
+
+public class RegressionReportHelper {
+
+    public static List<CaseResult> getAllCaseResultsForBuild(AbstractBuild build) {
+        List<CaseResult> ret = new ArrayList<CaseResult>();
+        List<AbstractTestResultAction> testActions = build.getActions(AbstractTestResultAction.class);
+
+        for (AbstractTestResultAction testAction : testActions) {
+            if (testAction instanceof TestResultAction) {
+                hudson.tasks.junit.TestResult testResult = (hudson.tasks.junit.TestResult) testAction.getResult();
+                ret.addAll(getTestsFromTestResult(testResult));
+            }
+            else if (testAction instanceof AggregatedTestResultAction) {
+                List<AggregatedTestResultAction.ChildReport> child_reports = ((AggregatedTestResultAction)testAction).getChildReports();
+                for (AggregatedTestResultAction.ChildReport child_report: child_reports) {

```

Comment:
Could you rename `child_report` to `childReport`? It would better to rename `child_reports` too.